streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -1163,684 +941,53 @@ public void shouldNotThrowExceptionOnScheduleIfCurrentNodeIsNotNull() {
-        }
-
-        assertTrue(producer.transactionCommitted());
-        assertFalse(producer.transactionInFlight());
-        assertFalse(producer.closed());
-    }
-
-    @Test
-    public void shouldAbortTransactionAndCloseProducerOnUncleanCloseWithEosEnabled() {
-        task = createStatelessTask(createConfig(true), StreamsConfig.METRICS_LATEST);
-        task.initializeTopology();
-
-        task.close(false, false);
-        task = null;
-
-        assertTrue(producer.transactionAborted());
-        assertFalse(producer.transactionInFlight());
-        assertTrue(producer.closed());
-    }
-
-    @Test
-    public void shouldAbortTransactionAndCloseProducerOnErrorDuringUncleanCloseWithEosEnabled() {
-        task = createTaskThatThrowsException(true);
-        task.initializeTopology();
-
-        task.close(false, false);
-
-        assertTrue(producer.transactionAborted());
-        assertTrue(producer.closed());
-    }
-
-    @Test
-    public void shouldOnlyCloseProducerIfFencedOnAbortDuringUncleanCloseWithEosEnabled() {
-        task = createStatelessTask(createConfig(true), StreamsConfig.METRICS_LATEST);
-        task.initializeTopology();
-        producer.fenceProducer();
-
-        task.close(false, false);
-        task = null;
-
-        assertTrue(producer.transactionInFlight());
-        assertFalse(producer.transactionAborted());
-        assertFalse(producer.transactionCommitted());
-        assertTrue(producer.closed());
-    }
-
-    @Test
-    public void shouldMigrateTaskIfFencedDuringFlush() {
-        final StateStore stateStore = new MockKeyValueStore(storeName, true, true);
-
-        final Map<String, String> storeToChangelogTopic = true ? Collections.singletonMap(storeName, storeName + "-changelog") : Collections.emptyMap();
-        final ProcessorTopology topology1 = new ProcessorTopology(
-            asList(source1, source2),
-            mkMap(mkEntry(topic1, source1), mkEntry(topic2, source2)),
-            Collections.emptyMap(),
-            singletonList(stateStore),
-            Collections.emptyList(),
-            storeToChangelogTopic,
-            Collections.emptySet()
-        );
-
-        task = new StreamTask(
-            taskId00,
-            partitions,
-            topology1,
-            consumer,
-            changelogReader,
-            createConfig(true),
-            streamsMetrics,
-            stateDirectory,
-            null,
-            time,
-            () -> producer = new MockProducer<>(false, bytesSerializer, bytesSerializer)
-        );
-        task.initializeStateStores();
-        task.initializeTopology();
-        producer.fenceProducer();
-
-        try {
-            task.flushState();
-            fail("Expected a TaskMigratedException");
-        } catch (final TaskMigratedException expected) {
-            assertThat(expected.migratedTask(), is(task));
-        }
-    }
-
-    @Test
-    public void shouldMigrateTaskIfFencedDuringProcess() {
```

Comment:
+1 I personally decided not to worry about it, but just keep a mental todo to come back after all the cleanups are complete to revisit test coverage (lines and semantics). But it would be nice to know why it's ok that they were removed.

Reply:
@abbccdda yeah my plan is to keep each PR against a separate `base` branch so that we do not need require each PR to be a "transaction" that always leave the branch still in a consistent and perfect state.

Reply:
I will add TODO to come back after all the cleanups for revamping our unit test coverage.